http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/http2/Http2MultiplexedChannelPool.java

```diff
@@ -124,6 +143,11 @@ private void releaseParentChannel0(Channel parentChannel, MultiplexedChannelReco
     }
 
     private void release0(Channel channel, Promise<Void> promise) {
+        if (closed) {
```

Comment:
Should we still let them release after close?

Reply:
Hmm not sure, I think it's better to error out so the caller can just close the channel if it gets rejected otherwise the other option would be to close it for them here but it seems a little non-obvious